src/e2e/init.rs

```diff
@@ -20,29 +22,6 @@ async fn test_init() -> Result<()> {
         },
     };
 
-    let init_payload = init.encode();
-
-    let gas_info = api
-        .calculate_upload_gas(
-            None,
-            gclient::code_from_os(PATH)?,
-            init_payload.clone(),
-            0,
-            true,
-        )
-        .await?;
-
-    let (message_id, _program_id, _hash) = api
-        .upload_program_bytes_by_path(
-            PATH,
-            gclient::now_micros().to_le_bytes(),
-            init_payload,
-            gas_info.min_limit,
-            0,
-        )
-        .await?;
-
-    assert!(listener.message_processed(message_id).await?.succeed());
-
+    let _invariant = init_invariant(&api, &mut listener, init).await;
```

Comment:
I think it's worth checking the values in the config fields as part of the init test, as well as ensuring that the remaining fields of the global struct are empty. For example, an empty list of fee tiers, etc.